control/controlhttp/server.go

```diff
@@ -75,12 +75,30 @@ func AcceptHTTP(ctx context.Context, w http.ResponseWriter, r *http.Request, pri
 	}
 	conn = netutil.NewDrainBufConn(conn, brw.Reader)
 
-	nc, err := controlbase.Server(ctx, conn, private, init)
+	cwc := newWriteCorkingConn(conn)
+
+	nc, err := controlbase.Server(ctx, cwc, private, init)
 	if err != nil {
 		conn.Close()
```

Comment:
you are not not calling `conn.Close` in all error paths

Reply:
nice, done, thanks